components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -3583,4 +3594,133 @@ protected String getSearchQuery(String searchQuery) throws APIManagementExceptio
         return criteria;
     }
 
+    @Override
+    public String getWSDLDocument(String username, String tenantDomain, String resourceUrl,
+            Map environmentDetails, Map apiDetails) throws APIManagementException {
+
+        if (username == null) {
+            username = APIConstants.END_USER_ANONYMOUS;
+        }
+        if (tenantDomain == null) {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+
+        Map<String, Object> docResourceMap = APIUtil.getDocument(username, resourceUrl, tenantDomain);
+        String wsdlContent = "";
+        if (log.isDebugEnabled()) {
+            log.debug("WSDL document resource availability: " + docResourceMap.isEmpty());
+        }
+        if (!docResourceMap.isEmpty()) {
+            try {
+                ByteArrayOutputStream arrayOutputStream = new ByteArrayOutputStream();
+                IOUtils.copy((InputStream) docResourceMap.get("Data"), arrayOutputStream);
+                String apiName = (String) apiDetails.get(API_NAME);
```

Comment:
Can't it be null?

Reply:
This is passed from the jaggery as a Map of <String, String>. It can be empty and handled in https://github.com/wso2/carbon-apimgt/pull/5054/files#diff-3dc01b68dc283f5c12cc5bf9e11fe4a8R3682